eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -464,29 +464,30 @@ setFreshDRepPulsingState epochNo stakePoolDistr epochState = do
       umap = dsUnified dState
       umapSize = Map.size $ umElems umap
       pulseSize = max 1 (umapSize `div` (fromIntegral :: Word64 -> Int) (4 * k))
-      epochState' =
-        epochState
-          & epochStateGovStateL . cgsDRepPulsingStateL
-            .~ DRPulsing
-              ( DRepPulser
-                  { dpPulseSize = pulseSize
-                  , dpUMap = dsUnified dState
-                  , dpIndex = 0 -- used as the index of the remaining UMap
-                  , dpStakeDistr = stakeDistr -- used as part of the snapshot
-                  , dpStakePoolDistr = stakePoolDistr
-                  , dpDRepDistr = Map.empty -- The partial result starts as the empty map
-                  , dpDRepState = vsDReps vState
-                  , dpCurrentEpoch = epochNo
-                  , dpCommitteeState = vsCommitteeState vState
-                  , dpEnactState =
-                      mkEnactState govState
-                        & ensTreasuryL .~ epochState ^. epochStateTreasuryL
-                  , dpProposals = proposalsActions props
-                  , dpProposalDeposits = proposalsDeposits props
-                  , dpGlobals = globals
-                  }
-              )
-  pure epochState'
+      govState' =
+        predictFuturePParams $
```

Comment:
So the cgsFuturePParams on which this function will operate at this point will always be `PotentialPParamsUpdate Nothing`  (because [we  are setting it like this in Epoch transition](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs#L341) ), correct?  
Also the `DRepPulsingState` will always be `DRPulsing`, so `extractDRepPulsingState` -> `finishDRepPulser` -> RATIFY rule etc will always be called when the thunk gets evaluated (presumably in the next `solidify`). 

(I don't have anything useful to say I'm afraid, just checking my understanding of the intention )

Reply:
> So the cgsFuturePParams on which this function will operate at this point will always be PotentialPParamsUpdate Nothing (because [we are setting it like this in Epoch transition](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs#L341) ), correct?

Correct.

> Also the DRepPulsingState will always be DRPulsing, so extractDRepPulsingState -> finishDRepPulser -> RATIFY rule etc will always be called when the thunk gets evaluated (presumably in the next solidify).

Also correct. By the time `solidify` is called, the DRep pulser will already be finished. If for some unusual reason it won't be finished, the call to `solidify` will force the pulser.